lua/legendary/compat/which-key.lua

```diff
@@ -17,7 +17,11 @@ end
 ---@param which_key_opts table
 ---@return LegendaryItem[]
 function M.parse_whichkey(which_key_tbls, which_key_opts)
-  local wk_parsed = require('which-key.keys').parse_mappings({}, which_key_tbls, which_key_opts.prefix or '')
+  local prefix = ''
+  if which_key_opts and which_key_opts.prefix then
+     prefix = which_key_opts.prefix
+  end
+  local wk_parsed = require('which-key.keys').parse_mappings({}, which_key_tbls, prefix)
```

Comment:
We can just do it this way, in a clean one-liner:

```suggestion
  local wk_parsed = require('which-key.keys').parse_mappings({}, which_key_tbls, which_key_opts and which_key_opts.prefix or '')
```